src/history/HistoryManagerImpl.cpp

```diff
@@ -97,7 +103,13 @@ writeCheckpointFile(Application& app, HistoryArchiveState const& has,
         app.getHistoryManager().isLastLedgerInCheckpoint(has.currentLedger));
     auto filename = publishQueueFileName(has.currentLedger);
     auto tmpOut = app.getHistoryManager().getTmpDir() / filename;
-    has.save(tmpOut.string());
+    {
+        OutputFileStream out(app.getClock().getIOContext(),
+                             /* fsyncOnClose */ true);
```

Comment:
Use `MODE_DISABLES_FSYNC` flag here instead of hardcoding to true.

Reply:
I purposely didn't want to do that: we need durable writes here regardless as to not break publish.

Reply:
maybe we should deprecate MODE_DISABLES_FSYNC. It's been enabled by default for a while, and is absolutely needed for critical paths like updating ledger state.

Reply:
We should keep it for testing, but I agree it shouldn't be exposed in prod paths. During my test refactor, i played around with fsync, and some tests ran literally 100x slower when it was enabled, so we definitely need to keep the config around and pass it around to streams correctly.

Reply:
sounds good, fixed (behind BUILD_TESTS)